### ISSUE TITLE
Cohort-based courses

### ISSUE THREAD

**Issue by nileshtrivedi** [learn-awesome/learn#241]
- A course will be a series of levels/challenges, curated by teachers which will be reputed topic experts
- Each level requires submitting a proof of learning
- Students can enroll in courses but they get access to the next level and the peer group ONLY after their proof of learning is accepted by the teachers
- The chat rooms for each level - supports highly engaging discussions, video calls, collaborations and more

This turns our existing syllabuses - which are sequences of learning resources along with commentary - into richer, more engaging courses. The fact that every level will require an accomplishment that students are genuinely proud of, will keep giving them the energy. And that they would be left out of amazing peer group unless they pass the levels, will keep giving them the motivation.

**Comment by nileshtrivedi** [comment]
- [ ] Upon logout from LearnAwesome, user should be logged out from chat as well
- [ ] Users can modify their nickname, but that will break their chat account link. Ideally, nickname change should update it in chat as well
- [ ] set_avatar API call is being made with admin token. It can be made with user's own token.
- [ ] Not sure what's the expiry policy for loginToken is. It may cause issues because we are saving it and always taking the saved value first.

**Comment by nileshtrivedi** [comment]
Enhancements:

- [ ] Allow multiple mentors for a course
- [ ] Paid courses

**Comment by prsmahajan** [comment]
Hi, I would like to work on this issue, Can I get started with this?

**Comment by nileshtrivedi** [comment]
@prsmahajan Yes. You can go through the commits above, but before you start adding new code, propose a plan here first?

**Comment by prsmahajan** [comment]
What are the problems you are facing?

**Comment by nileshtrivedi** [comment]
There are unchecked items listed in my comment above: https://github.com/learn-awesome/learn/issues/241#issuecomment-812982028

But perhaps, you can suggest your own improvements once you try the Course feature.

**Comment by prsmahajan** [comment]
(Upon logout from LearnAwesome, user should be logged out from chat as well
 Users can modify their nickname, but that will break their chat account link. Ideally, nickname change should update it in chat as well) I can resolve these two issues, Would you like to assign me for this issue?

**Comment by nileshtrivedi** [comment]
Can you ping me on [LearnAwesome slack](https://learnawesome.org/join_slack)? Let's discuss a bit.

**Comment by sahilcode17** [comment]
hi, @nileshtrivedi I would like to contribute to this issue, Is it still open or have any changes been made further. Do let me know
